index.js

```diff
@@ -505,14 +511,70 @@ class XiaomiRoborockVacuum {
         await this.device.activateCleaning();
       } else if (!state) { // Stop cleaning
         this.log.info(`ACT setCleaning | ${this.model} | Stop cleaning and go to charge.`);
-        await this.device.activateCharging(); // Charging works for 1st, not for 2nd
+        await this.activateCharging(); // Charging works for 1st, not for 2nd
       }
     } catch (err) {
       this.log.error(`ERR setCleaning | ${this.model} | Failed to set cleaning to ${state}`, err);
       throw err;
     }
   }
 
+  async setCleaningRoom(state, room) {
+    await this.ensureDevice('setCleaning');
+
+    try {
+      if (state && !this.isCleaning) { // Start cleaning
+        this.log.info(`ACT setCleaning | ${this.model} | Start cleaning Room ID ${room}, not charging.`);
+        await this.device.call('app_segment_clean', [room]);
+      } else if (!state) { // Stop cleaning
+        this.log.info(`ACT setCleaning | ${this.model} | Stop cleaning and go to charge.`);
+        await this.activateCharging();
+      }
+    } catch (err) {
+      this.log.error(`ERR setCleaning | ${this.model} | Failed to set cleaning to ${state}`, err);
+      throw err;
+    }
+  }
+
+  async activateCharging() {
+    await this.ensureDevice('activateCharging');
+    try {
+      await this.device.call('app_charge');
```

Comment:
if keeping it in here (nothing against that, but check discussion https://github.com/nicoh88/homebridge-xiaomi-roborock-vacuum/pull/113/files#r383563234), make sure the response from the robot is `ok` and not an error: something like
https://github.com/nicoh88/homebridge-xiaomi-roborock-vacuum/blob/2005257a6efb0e6a70c7ebcf41ef77018c34d6d1/index.js#L695-L700

Reply:
yes will look at it in next PR 👍